Tombstoning

NickolausDS opened the pull request:
Support features created here: https://github.com/fair-research/fair-identifiers-client/pull/3

There are a couple oddities in this PR. A summary is: 

* Default args were being passed and overwriting values
    * See here for an explanation: https://github.com/fair-research/minid/commit/9c3409a66b78ce83646a6a3339b385f58d2305a2
*  update `active` cannot be passed directly, due to ambiguity. From the CLI, `--set-active` or `--set-inactive` will explicitly change this value. In the SDK, active isn't passed to the identifiers client if it shouldn't change. It's only explicitly passed down if given, to avoid a default automatically changing the value when it shouldn't. 

There may be better solutions here. One of the repercussions of not passing values is they cannot be nullified. For example, if someone accidentally sets 'replaced-by' for the wrong minid, there is no way to take off the erroneous 'replaced-by'. One possible fix here is setting a magic value 'None' or 'unset' to explicitly pass 'None' to clear an existing value. Example:
    * `$ minid update minid:123 --replaced-by None`

NickolausDS committed:
Added new/old minid 'tombstoning' features

NickolausDS committed:
Made --set-active and --set-inactive different args

NickolausDS committed:
Fixed update not translating minid to hdl

NickolausDS committed:
Fixed default None values being sent to the server

NickolausDS committed:
Added code docs to register

NickolausDS committed:
Tombstoned args translated from minid to hdl if supplied

NickolausDS committed:
Added tests

NickolausDS committed:
Fixed unintenedd args being passed to update

Update has an odd problem. It, like many other functions, allowed
default args if they were not given. However, the default args
generated had a tendency to clear existing values. For example,
running the following:

minid update minid:123 --replaces minid:456

Would end up clearing the network locations for minid:123, since
update would end up passing locations=[] as a default args.

The change here includes not passing args if they have a falsey
value. This disallowes explicitly clearing values, but solves the
problem above.

NickolausDS committed:
Fixed python 3.6 test

mikedarcy reviewed (approved):
Looks good. I agree that using `None` to "unset" `replaces` and `replaced_by` is useful.

NickolausDS committed:
Added feature to clear existing replaced, replaced-by minids

NickolausDS committed:
Fixed inactive minids from displaying status

NickolausDS committed:
Fixed metadata not always being updated,

NickolausDS committed:
Allow None to be specified for locations

NickolausDS committed:
Added tests for minid commands

NickolausDS committed:
Fixed date parsing for python 3.6

NickolausDS commented:
@mikedarcy The earlier problem of not being able to clear values for the `replaced-by` and `replaces` has been fixed, along with some other bugs. For a few different values (locations, replaced, replaced-by) the special keyword string 'None' can be passed on the CLI to clear existing values.

```
$ minid update minid.test:7TDW9m5bRh5x --replaces None
```

On the sdk, it's the same except the value is a regular Nonetype. 
```
minid_client.update('minid.test:123', replaces=None)
```

**NOTE**: The Identifier Service doesn't yet support unsetting values, so this isn't quite finished.

mikedarcy reviewed (approved):
Please merge and I will work on the back end changes required.
